actors/common/src/main/java/org/sunbird/learner/util/Util.java

```diff
@@ -939,6 +939,7 @@ public static void initializeContext(Request actorMessage, String env) {
       requestContext.put(JsonKey.ENV, env);
       requestContext.put(JsonKey.REQUEST_ID, actorMessage.getRequestId());
       requestContext.put(JsonKey.REQUEST_TYPE, JsonKey.API_CALL);
+      requestContext.put(JsonKey.DEVICE_ID, actorMessage.getContext().get(JsonKey.DEVICE_ID));
```

Comment:
actorMessage.getContext() can be actorMessage.getContext() != null ? actorMessage.getContext().get(JsonKey.DEVICE_ID): "NA"